File: backend/src/main/java/com/prepkeeper/entity/EasterEgg.java
package com.prepkeeper.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;

/**
 * 彩蛋触发记录实体
 */
@Data
@Entity
@Table(name = "easter_eggs")
public class EasterEgg {

    @Id
    @Column(name = "record_id", length = 64)
    private String recordId;

    @Column(name = "user_id", length = 64, nullable = false)
    private String userId;

    /**
     * 彩蛋类型: late_night/weekend/early_bird/random
     */
    @Column(name = "egg_type", length = 50, nullable = false)
    private String eggType;

    @Column(name = "trigger_date", nullable = false)
    private LocalDate triggerDate;

    @Column(name = "content", length = 500)
    private String content;

    /**
     * 是否已触发: 0-否 1-是
     */
    @Column(name = "is_triggered")
    private Integer isTriggered = 0;
}

File: agent-service/app/models/easter_egg.py
"""
彩蛋触发记录模型
"""
from sqlalchemy import Column, String, Integer, Date, TIMESTAMP, SmallInteger, ForeignKey, Text
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.models.database import Base
import uuid


class EasterEgg(Base):
    """彩蛋触发记录表"""
    __tablename__ = "easter_eggs"

    record_id = Column(String(64), primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id = Column(String(64), ForeignKey("users.user_id"), nullable=False)
    egg_type = Column(String(50), nullable=False)  # late_night/weekend/early_bird/random
    trigger_date = Column(Date, nullable=False)
    content = Column(String(500), nullable=True)
    is_triggered = Column(SmallInteger, default=0)  # 0-否 1-是

    # 关系
    user = relationship("User", backref="easter_eggs")

    def __repr__(self):
        return f"<EasterEgg(record_id={self.record_id}, type={self.egg_type})>"